loki/transformations/utilities.py

```diff
@@ -623,20 +653,98 @@ def find_driver_loops(section, targets):
         List of subroutines that are to be considered as part of
         the transformation call tree.
     """
+    class FindDriverLoops:
```

Comment:
Should this maybe inherit from `loki.ir.GenericVisitor`? In fact, building a whole new type of visitor here seems to introduce quite some complexity, and I'm not quite seeing why that is strictly necessary?